webview/src/connect/components/Studio.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react'
+import { MessageFromWebviewType } from 'dvc/src/webview/contract'
+import { STUDIO_URL } from 'dvc/src/connect/webview/contract'
+import { EmptyState } from '../../shared/components/emptyState/EmptyState'
+import { Button } from '../../shared/components/button/Button'
+import { sendMessage } from '../../shared/vscode'
+
+export const Studio: React.FC = () => {
+  const openStudio = () =>
+    sendMessage({ type: MessageFromWebviewType.OPEN_STUDIO })
+
+  const openStudioProfile = () =>
+    sendMessage({ type: MessageFromWebviewType.OPEN_STUDIO_PROFILE })
+
+  const saveStudioToken = () =>
+    sendMessage({ type: MessageFromWebviewType.SAVE_STUDIO_TOKEN })
```

Comment:
Since none of these functions rely on props on state, I'd move them outside of the component as these get recreated on each re-render. This should fix the 2 codeclimate issues as well.

Reply:
🙏🏻